assets/js/stripe-payment-request.js

```diff
@@ -649,7 +653,11 @@ jQuery( function( $ ) {
 							displayItems: response.displayItems,
 						} )
 					).then( function () {
-						$( document.body ).trigger( 'wc_stripe_unblock_payment_request_button' );
+						if (response.is_in_stock) {
+							$( document.body ).trigger( 'wc_stripe_unblock_payment_request_button' );
+						} else {
+							$( document.body ).trigger( 'wc_stripe_hide_payment_request_button' );
```

Comment:
I think it's better if we use `wc_stripe_payment_request.unblockPaymentRequestButton()` directly here instead of registering and firing a custom jQuery event. In the case if `wc_stripe_unblock_payment_request_button` we created an event with that name because we wanted 3rd party extensions to be able to trigger the event.